src/olympia/search/tests/test_search_ranking.py

```diff
@@ -0,0 +1,566 @@
+            'Open image in a new tab',
+            'Open Image in New Tab',
+        ))
+
+    def test_scenario_coinhive(self):
+        # TODO, should match "CoinBlock"
+        self._check_scenario('CoinHive', (
+            'Coinhive Blocker',
+            'NoMiners',  # via description
+            # 'CoinBlock',  # via prefix search
+        ))
+
+    def test_scenario_privacy(self):
+        self._check_scenario('Privacy', (
+            'Privacy Badger',
+            'Privacy Settings',
+            'Google Privacy',  # More users, summary
+            'Privacy Pass',
+            'Ghostery',  # Crazy amount of users, summary
+            'Blur',  # summary + many users but not as many as ghostery
+        ))
+
+    def test_scenario_firebu(self):
+        self._check_scenario('firebu', (
+            'Firebug',
+            # unclear why preference to Firebug Autocompleter,
+            # weekly downloads + users?
+            'Firefinder for Firebug',
+            'Firebug Autocompleter',
+            'Fire Drag',
+        ))
+
+    def test_scenario_fireb(self):
+        self._check_scenario('fireb', (
+            'Firebug',
+            'Firefinder for Firebug',
+            'Firebug Autocompleter',
+            'Fire Drag',
+        ))
+
+    def test_scenario_menu_wizzard(self):
+        self._check_scenario('Menu Wizzard', (
+            'Menu Wizard',  # (fuzzy, typo)
+            'Add-ons Manager Context Menu',  # partial match + users
+        ))
+
+    def test_scenario_frame_demolition(self):
+        self._check_scenario('Frame Demolition', (
+            'Frame Demolition',
+        ))
+
+    def test_scenario_demolition(self):
+        # Find "Frame Demolition" via a typo
+        self._check_scenario('Demolation', (
+            'Frame Demolition',
+        ))
+
+    def test_scenario_restyle(self):
+        self._check_scenario('reStyle', (
+            'reStyle',
+        ))
+
+    def test_scenario_megaupload_downloadhelper(self):
+        # Doesn't find "RapidShare DownloadHelper" anymore
+        # since we now query by "MegaUpload AND DownloadHelper"
+        self._check_scenario('MegaUpload DownloadHelper', (
+            'MegaUpload DownloadHelper',
+        ))
+
+    def test_scenario_downloadhelper(self):
+        # No direct match, "Download Flash and Video" has
+        # huge amount of users that puts it first here
+        self._check_scenario('DownloadHelper', (
+            'Download Flash and Video',
+            '1-Click YouTube Video Download',
+            'RapidShare DownloadHelper',
+            'MegaUpload DownloadHelper',
+        ))
+
+    def test_scenario_megaupload(self):
+        self._check_scenario('MegaUpload', (
+            # I have litterally NO idea :-/
+            'Popup Blocker',
+            'MegaUpload DownloadHelper',
+        ))
+
+    def test_scenario_no_flash(self):
+        # TODO: Doesn't put "No Flash" on first line, does the "No"
+        # do something special here?
```

Comment:
`no` is a stopword FWIW

Reply:
That's true but the "no idea" part comes where we have a raw term match and IMHO (correct me if I'm wrong) that raw term match should not make use of any analyzing, stopword filtering etc